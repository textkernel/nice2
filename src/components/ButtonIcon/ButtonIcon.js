import React, { Children } from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './ButtonIcon.scss';
import { CONTEXTS } from '../../constants';

const { block } = bem({
    name: 'ButtonIcon',
    classnames: styles,
    propsToMods: ['context', 'isActive']
});

const ButtonIcon = props => {
    const { children, ...rest } = props;

    return (
        <button {...rest} {...block(props)}>
            {Children.only(children)}
        </button>
    );
};

ButtonIcon.displayName = 'ButtonIcon';

ButtonIcon.propTypes = {
    context: PropTypes.oneOf([...CONTEXTS]),
    isActive: PropTypes.bool
};

ButtonIcon.defaultProps = {
    context: 'neutral',
    isActive: false
};

export default ButtonIcon;
