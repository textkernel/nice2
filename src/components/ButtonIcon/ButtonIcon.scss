@import '../../themes/oneui/oneui.scss';

.ButtonIcon {
    background-color: transparent;
    border: none;
    outline: none;
    color: var(--color-foreground);
    line-height: 36px;
    height: 36px;
    width: 36px;
    display: inline-flex;
    font-size: var(--font-size-normal);
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    cursor: pointer;
    transition: all var(--transition-duration);

    &:hover {
        background-color: var(--color-neutral-10);
    }

    &--context {
        @each $context in ($contexts) {
            &_#{ $context } {
                color: var(--color-#{$context});

                &:hover {
                    background-color: var(--color-#{$context});
                    color: var(--color-background);
                }

                &:active,
                &.ButtonIcon--isActive {
                    background-color: var(--color-#{$context}-60);
                    color: var(--color-background);
                }
            }
        }
    }
}
