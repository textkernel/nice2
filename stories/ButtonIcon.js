import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, select, withKnobs } from '@storybook/addon-knobs';
import { ButtonIcon } from '@textkernel/oneui';
import { CONTEXTS } from '../src/constants';

storiesOf('ButtonIcon', module)
    .addDecorator(withKnobs)
    .add('ButtonIcon', () => (
        <ButtonIcon
            context={select('Context', CONTEXTS, CONTEXTS[0])}
            isActive={boolean('isActive', false)}
        >
            <i className="far fa-bookmark" />
        </ButtonIcon>
    ));
