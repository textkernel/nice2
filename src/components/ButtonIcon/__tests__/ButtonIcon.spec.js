import React from 'react';
import toJson from 'enzyme-to-json';
import ButtonIcon from '../ButtonIcon';

describe('<ButtonIcon> that renders a button icon', () => {
    it('should render a default progress bar correctly', () => {
        const wrapper = shallow(
            <ButtonIcon>
                <i className="far fa-bookmark" />
            </ButtonIcon>
        );
        expect(toJson(wrapper)).toMatchSnapshot();

        wrapper.setProps({
            isActive: true
        });

        expect(wrapper.hasClass('ButtonIcon--isActive')).toBe(true);

        wrapper.setProps({
            context: 'brand'
        });

        expect(wrapper.hasClass('ButtonIcon--context_brand')).toBe(true);
    });
});
