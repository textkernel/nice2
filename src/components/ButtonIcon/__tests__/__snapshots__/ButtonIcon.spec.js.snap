// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonIcon> that renders a button icon should render a default progress bar correctly 1`] = `
<button
  className="ButtonIcon ButtonIcon--context ButtonIcon--context_neutral"
  context="neutral"
  isActive={false}
>
  <i
    className="far fa-bookmark"
  />
</button>
`;
